Sources/TuistAutomation/XcodeBuild/XcodeBuildController.swift

```diff
@@ -61,7 +62,11 @@ public final class XcodeBuildController: XcodeBuildControlling {
         // Destination
         switch destination {
         case let .device(udid):
-            command.append(contentsOf: ["-destination", "id=\(udid)"])
+            var value = ["id=\(udid)"]
+            if rosetta {
+                value += ["arch=x86_64"]
+            }
+            command.append(contentsOf: ["-destination", value.joined(separator: ",")])
```

Comment:
```suggestion
            command.append(contentsOf: ["-destination", "id=\(udid)"])
            if rosetta {
                command.append("arch=x86_64")
            }
```

Reply:
The value of destination must be separated by comma.